### ISSUE TITLE
Uncaught TypeError: Cannot read property 'push' of undefined

### ISSUE THREAD

**Issue by HeavenSky** [gaearon/react-hot-loader#1041]
If you are reporting a bug or having an issue setting up React Hot Loader, please fill in below. For feature requests, feel free to remove this template entirely.

### Description
when I use `react-hot-loader@4` in `webpack@1` `react@0`, change a component, it will report error in browser console, and [HMR] failed. detail error is below.

What you are reporting:
```
react-hot-loader.development.js:1057 Uncaught TypeError: Cannot read property 'push' of undefined
    at hydrateLegacyStack (react-hot-loader.development.js:1057)
    at hydrateLegacyStack (react-hot-loader.development.js:1056)
    at react-hot-loader.development.js:1061
    at Array.forEach (<anonymous>)
    at hydrateLegacyStack (react-hot-loader.development.js:1059)
    at hydrateLegacyStack (react-hot-loader.development.js:1056)
    at react-hot-loader.development.js:1061
    at Array.forEach (<anonymous>)
    at hydrateLegacyStack (react-hot-loader.development.js:1059)
    at hydrateLegacyStack (react-hot-loader.development.js:1056)
```

### Expected behavior

What you think should happen: when i change component, [HMR] should success

### Actual behavior

What actually happens: [HMR] failed and reported an error in browser console.

### Environment

React Hot Loader version:  4.3.4

Run these commands in the project folder and fill in their results:

1.  `node -v`: v8.11.3
2.  `npm -v`:  6.2.0

Then, specify:

1.  Operating system:  MAC OSX 10.13.6
2.  Browser and version: chrome 67.0.3396.99

### Reproducible Demo
https://github.com/HeavenSky/webpack-frames/tree/issue/webpack1-ie8
Above is the issue demo code.
    1. start server `npm i`  `npm ddp`  `npm start`
    2. open chrome url:   http://localhost:8888/
    3. open browser console
    4. change `src/containers/App.js` component,  browser console will report error.

I try to fix this issue. edit `node_modules/react-hot-loader/dist/react-hot-loader.development.js`, LineNumber: 1057
```js
stack.children.push(childStack);  // change to `stack.children && stack.children.push(childStack);` this issue is not happened, [HMR] success.
```

Please take the time to create a new project that reproduces the issue.

You can copy your project that experiences the problem and start removing things until you’re left with the minimal reproducible demo. This helps contributors, and you might get to the root of your problem during that process.

Push to GitHub and paste the link here.

**Comment by theKashey** [comment]
That's super interesting!

**Comment by theKashey** [comment]
React-Hot-Loader v4 does not support React14.
New RHL == New React. Version 15 at least.

**Comment by HeavenSky** [comment]
@theKashey
My temporary solution is
```js
const F = "node_modules/react-hot-loader/dist/react-hot-loader.development.js";
if (fs.existsSync(dir(F))) {
	let str = fs.readFileSync(dir(F), "utf-8");
	str = str.replace(/stack.children.push/g, '(stack.children||[]).push');
	fs.writeFileSync(dir(F), str, "utf-8");
}
```

**Comment by theKashey** [comment]
It will not work. Let me explain.
1. It breaks because `stack.children.push(childStack);` is not defined.

Why it is not defined?
```js
function hydrateLegacyStack(node, stack) {
  if (node._currentElement) {
    pushState(stack, node._currentElement.type, node._instance || stack)
    // ^^^ cos this path was not executed
  }

  if (node._renderedComponent) {
    const childStack = {} // <--- and "stack" is an empty object
```
So - you are fixing runtime error, but `stack` __keeps being an empty object__.

I've opened a [PR](https://github.com/gaearon/react-hot-loader/pull/1042), which adds children to the `stack`, not ignoring them as you propose. And that reveal another problem - [hotReplacementRender](https://github.com/gaearon/react-hot-loader/blob/master/src/reconciler/hotReplacementRender.js)(or "ReactHotLoader v4") ignores objects without `type` set, as long have to idea how to "render" them,  while `stack` has only `children` or even nothing, in your case.

In short - React-Hot-Loader v4 is NOT working for React 15. If it works for you - then React-Hot-Loader __v3__ works for you, but everything that distinguishes v4 from v3 - does not.

For old stuff, like webpack 1 or React 14 you should __use older versions of RHL__, which does not rely on React15/16 internals, as RHL v4 does.

**Comment by HeavenSky** [comment]
@theKashey   I got it.   Thanks very much for your patient answer.
